rotkehlchen/usage_analytics.py

```diff
@@ -22,134 +24,114 @@
-    )
-
-
-def query_ipinfo() -> Optional[GeolocationData]:
-    """
-    50,000 requests per month tied to the API Key
-    https://ipinfo.io/developers
-    """
-    try:
-        response = requests.get(
-            'https://ipinfo.io/json?token=16ab40aad9bd5b',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
+        data = response.json()
+        date = deserialize_timestamp_from_date(
+            date=data['commit']['commit']['author']['date'],
+            formatstr='%Y-%m-%dT%H:%M:%S',
+            location='Analytics',
         )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
-        return None
-
+    except requests.exceptions.RequestException as e:
+        log.debug(f'Failed to get metadata information for geoip file. {str(e)}')
+        metadata_query_failed = True
+    except DeserializationError as e:
+        log.debug(f'Failed to deserialize date in metadata information for geoip file. {str(e)}')
+        metadata_query_failed = True
+    except KeyError as e:
+        log.debug(f'Github response for geoip file had missing key {str(e)}')
+        metadata_query_failed = True
+
+    if metadata_query_failed:
+        old_files = glob.glob(str(geoip_dir) + '/*.mmdb')
+        if len(old_files) == 0:
+            return None
+        filename = geoip_dir / old_files[0]
+    else:
+        filename = geoip_dir / f'geoip-{date}.mmdb'
+
+    if not filename.is_file():
+        # Remove old files
+        files = glob.glob(str(geoip_dir) + '/*')
+        for f in files:
+            os.remove(f)
+        # Download latest version
+        try:
+            response = requests.get(
+                url='https://github.com/geoacumen/geoacumen-country/raw/master/Geoacumen-Country.mmdb',  # noqa: E501
+                timeout=DEFAULT_TIMEOUT_TUPLE,
+                stream=True,
+            )
+        except requests.exceptions.RequestException as e:
+            log.debug(f'Failed to download geoip database file. {str(e)}')
+            return None
+        with open(filename, 'wb') as outfile:
+            response.raw.decode_content = True
+            shutil.copyfileobj(response.raw, outfile)
+
+    # get user ip
     try:
-        json_ret = jsonloads_dict(response.text)
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def query_ipstack() -> Optional[GeolocationData]:
-    """
-    10,000 requests per month tied to the API Key
-    https://ipstack.com/
-    """
-    try:
-        response = requests.get(
-            'http://api.ipstack.com/check?access_key=affd920d6e1008a614900dbc31d52fa6',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
-        )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
+        user_ip = requests.get(
```

Comment:
We should also fix this. There must be a better way to get the ip of the user than checking an external service.

I know, I know ... you can only reliably find the LAN ip and no non-external solution is perfect. I think we can use miniupnp for this. Which tries to query the user's router. If that fails I think it's fair to simply not use a location in usage analytics.

Example here: https://stackoverflow.com/a/41385033/110395

You should study how miniupnpc works and also do proper error checking. In all errors just don't send location in usage analytics.